src/main/java/org/truffleruby/core/module/ModuleFields.java

```diff
@@ -708,6 +708,11 @@ public SourceSection getSourceSection() {
         return sourceSection;
     }
 
+    /**
+     * Registers an Assumption for a given method name, which is invalidated when a method with same
+     * name is defined or undefined in this class or when a module is prepended to this class.
```

Comment:
Is this accurate? I thought the assumption was invalidated when any change is made to the class.

Reply:
Yes, look at the code below.
The normal per-class/module Assumption is a different thing and invalidated on every method def/undef on that class.